### ISSUE TITLE
Migrate Provisioner to the new Director API 

### ISSUE THREAD

**Issue by franpog859** [kyma-project/control-plane#1598]
**Why to do**

Compass plan (info from 13.04.2022) to deprecate the old Director API (direct management of scenario labels and label definitions) but they can do that only if all of the clients are properly migrated to the new one

The deadline: Q2 or early Q3 (info from 27.04.2022) - **We plan to remove Provisioner as a whole component soon so I'd suggest just waiting for the removal. Consider this issue as _has to be done_ only if the migration deadline is shorter**

**What to do**

Migrate [Provisioner](https://github.com/kyma-project/control-plane/blob/main/components/provisioner/internal/director/query.go) to the new Compass Director API.

List of mutations replacements to be done based on the discussion from 14.04.2022 with @DimitarPetrov :
- `registerRuntime` with scenario label - In the future scenario label may be treated as protected and modified only via formation APIs. You can register the runtime and then assign it to a formation in separate calls.
- `updateRuntime` - I assume that there is also the same problem with the scenario labels. Update scenarios only via formation APIs in a separate call, use the [`setRuntimeLabel` mutation](https://github.com/kyma-incubator/compass/blob/main/components/director/pkg/graphql/schema.graphql#L1412)

The new API is concurrent safe and responds with the error if one occurred

You can find the new API [here](https://github.com/kyma-incubator/compass/blob/main/components/director/pkg/graphql/schema.graphql#L1379)

**Comment by DimitarPetrov** [comment]
Minor remark regarding `updateRuntime` mutation. We plan to remove the labels from updateRuntime input.

If scenario label is modified -> use [(un)assignFormation](https://github.com/kyma-incubator/compass/blob/main/components/director/pkg/graphql/schema.graphql#L1439)

If another label is modified -> use [setRuntimeLabel](https://github.com/kyma-incubator/compass/blob/main/components/director/pkg/graphql/schema.graphql#L1469) (scenario label will be forbidden in this mutation)

**Comment by franpog859** [comment]
Instead of registering runtime with the scenario labels all in one Director call, we should do it in two steps: register runtime and then add the runtime to the scenario

**Comment by akgalwas** [comment]
There is no need for updating `registerRuntime` calls. Provisioner does set Labels in  Provisioner but it doesn't manage scenarios so the existent code will work correctly. Mind that Kyma Environment Broker passes the following Label

**Comment by koala7659** [comment]
I have checked the source code and I fully agree with @akgalwas. Latest updates to Director's API have no impact on how Provisioner calls should be made. Director is called only to registered the runtime with KEB labels, but NO scenario label is involved. Provisioner does not register runtime for scenario. It must be done for example on the global account level in BTP, and provisioner is not involved there.

Tested updated Provisioner version with latest Director dependency for provisioning/deprovisioning/shoot upgrade. Works fine.
